src/client/index.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable no-console */
+/* eslint-disable no-undef */
+const ws = new WebSocket('ws://localhost:8080');
+/* eslint-enable no-undef */
+let currentRoom = null;
+let currentRoomName = '';
+
+ws.onmessage = (event) => {
+  const data = JSON.parse(event.data);
+
+  switch (data.type) {
+    case 'roomCreated':
+      console.log(`Room created with ID: ${data.roomId}`);
+      break;
+
+    case 'roomJoined':
+      currentRoom = data.roomId;
+      currentRoomName = data.roomName || document.getElementById('room').value;
+
+      document.getElementById('chat').innerHTML =
+        `<h2>Messages in <span id="roomName">${currentRoomName}</span> chat:</h2>`;
+      data.messages.forEach((msg) => addMessage(msg));
+      enableChat();
+      document.getElementById('roomControls').style.display = 'block';
+      break;
+
+    case 'newMessage':
+      addMessage(data.message);
+      break;
+
+    case 'roomRenamed':
+      currentRoomName = data.newName;
+      document.getElementById('roomName').textContent = currentRoomName;
+      console.log(`Room renamed to: ${data.newName}`);
+      break;
+
+    case 'roomDeleted':
+      if (currentRoom === data.roomId) {
+        console.log('Room deleted, please join another room');
+        currentRoom = null;
+        currentRoomName = '';
+        document.getElementById('chat').innerHTML = '';
+        disableChat();
+        document.getElementById('roomControls').style.display = 'none';
+      }
+      break;
+  }
+};
+
+function setUsername() {
+  const username = document.getElementById('username').value;
+
+  if (!username) {
+    console.log('Please enter a username.');
+
+    return;
+  }
+  /* eslint-disable no-undef */
+  localStorage.setItem('username', username);
+  /* eslint-enable no-undef */
+
+  ws.send(JSON.stringify({ type: 'setUsername', username }));
```

Comment:
Make sure the server is set up to handle the 'setUsername' message type. Otherwise, this action will not have any effect.